src/middlewared/middlewared/plugins/crypto.py

```diff
@@ -184,9 +130,541 @@ def _validate_certificate_with_key(certificate, private_key, schema_name, verror
             'Please provide a valid csr_id which has a valid CSR filed'
         )
 
-    await middleware.run_in_thread(
-        _validate_certificate_with_key, certificate, private_key, schema_name, verrors
+    await middleware.call(
+        'cryptokey.validate_certificate_with_key', certificate, private_key, schema_name, verrors, passphrase
+    )
+
+    key_type = data.get('key_type')
+    if key_type:
+        if key_type != 'EC' and not data.get('key_length'):
+            verrors.add(
+                f'{schema_name}.key_length',
+                'This field is required with RSA based keys'
```

Comment:
```suggestion
                'RSA-based keys require an entry in this field.'
```
(Although this does not really tell what type of entry.)

Reply:
`f'{schema_name}.key_length'` defines the type of entry. Key length are integers by type. Please advise how should we proceed here